src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,225 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import cx from 'classnames';
+
+export const TreeContext = React.createContext<
+  | {
+      selectedNodes?: Array<string>;
+      setSelectedNode?: (
+        node: Array<string> | ((prev: Array<string>) => Array<string>),
+      ) => void;
+      expandedNodes?: Array<string>;
+      setExpandedNodes?: (
+        node: Array<string> | ((prev: Array<string>) => Array<string>),
+      ) => void;
+      nodeDepth?: number;
+      parentNode?: FlatNode;
+    }
+  | undefined
+>(undefined);
+
+export type NodeData = {
+  subnodes: Array<NodeData>;
+  nodeId: string;
+  label?: string;
+  subLabel?: string;
+  isExpanded?: boolean;
+  isDisabled?: boolean;
+};
+
+type FlatNode = {
+  data: NodeData;
+  id: string;
+  depth: number;
+  subNodeIds: string[];
+  parent?: FlatNode;
+};
+
+export type TreeProps = {
+  /**
+   * Node renderer.
+   */
+  nodeRenderer?: (props: NodeData) => JSX.Element;
```

Comment:
If node rendered is not provided by user, nodes are not being displayed. Maybe make it mandatory parameter or give it default value?

Reply:
Added default value for nodeRenderer

Reply:
Current default nodeRenderer is not functional at all, user needs to provide their own renderer if they want ANY functions (expand, select, etc). My idea for default renderer was to have it same as in Basic story example so that story wouldn't have any renderer provided and add another story with "custom" renderer (eg. visibility checkbox added). Let me know if this makes sense.

Reply:
node renderer is now a required prop